app/src/org/commcare/views/widgets/MediaWidget.java

```diff
@@ -237,21 +235,12 @@ public void setBinaryData(Object binaryuri) {
             }
         }
 
-        if (newMedia.exists()) {
-            // Add the copy to the content provider
-            ContentValues values = new ContentValues(6);
-            values.put(MediaStore.Audio.Media.TITLE, newMedia.getName());
-            values.put(MediaStore.Audio.Media.DISPLAY_NAME, newMedia.getName());
-            values.put(MediaStore.Audio.Media.DATE_ADDED, System.currentTimeMillis());
-            values.put(MediaStore.Audio.Media.DATA, newMedia.getAbsolutePath());
-
-            Uri mediaUri =
-                    getContext().getContentResolver().insert(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, values);
-            String mediaUriString = mediaUri == null ? "null" : mediaUri.toString();
-            Log.i(TAG, "Inserting media returned uri = " + mediaUriString);
+        try {
+            FileUtil.addMediaToGallery(getContext(), newMedia);
             showToast("form.attachment.success");
-        } else {
+        } catch (Exception e) {
```

Comment:
We should let the app crash in case of exceptions here if we are going to retain this functionality.